src/com/theironyard/novauc/Main.java

```diff
@@ -1,8 +1,173 @@
+        stmt.execute("CREATE TABLE IF NOT EXISTS posts (id IDENTITY, postId INT, text VARCHAR, user_name VARCHAR)");
+    }
+
+//    public static int lengthOfTable(Connection conn, String currentUser) throws SQLException{
+//        int id=1;
+//        PreparedStatement stmt = conn.prepareStatement("SELECT COUNT(*) FROM posts INNER JOIN users ON posts.user_name = users.user_name WHERE users.user_name = ?");
+//        stmt.setString(1, currentUser);
+//        ResultSet results = stmt.executeQuery();
+//        if (results.next()) {
+//            id = results.getInt("COUNT(*)")+1;
+//        }
+//        return id;
+//    }
+
+    public static void insertUser(Connection conn, String name, String password) throws SQLException {
+        PreparedStatement stmt = conn.prepareStatement("INSERT INTO users VALUES (NULL, ?, ?)");
+        stmt.setString(1, name);
+        stmt.setString(2, password);
+        stmt.execute();
+    }
+
+    public static User selectUser(Connection conn, String name) throws SQLException {
+        PreparedStatement stmt = conn.prepareStatement("SELECT * FROM users WHERE user_name = ?");
+        stmt.setString(1, name);
+        ResultSet results = stmt.executeQuery();
+        if (results.next()) {
+            int id = results.getInt("id");
+            String password = results.getString("password");
+            return new User(id, name, password);
+        }
+        return null;
+    }
+
+
+    public static void insertMessage(Connection conn, String text,String userName) throws SQLException {
+        PreparedStatement stmt = conn.prepareStatement("INSERT INTO posts VALUES (NULL, ?, ?, ?)");
+        stmt.setInt(1,idnumber);
+        stmt.setString(2, text);
+        stmt.setString(3, userName);
+        idnumber++;
+        stmt.execute();
+    }
+
+    public static void deleteMessage(Connection conn,int id) throws SQLException{
+        PreparedStatement stmt = conn.prepareStatement("DELETE FROM posts WHERE posts.postId = ?");
+        stmt.setInt(1, id);
+        stmt.execute();
+    }
+
+    public static ArrayList<Posts> selectAllMessages(Connection conn,String currentUser) throws SQLException {
+        ArrayList<Posts> messages = new ArrayList<>();
+        PreparedStatement stmt = conn.prepareStatement("SELECT * FROM posts INNER JOIN users ON posts.user_name = users.user_name WHERE users.user_name = ?");
+        stmt.setString(1,currentUser);
+        ResultSet results = stmt.executeQuery();
+        while (results.next()) {
+            int id = Integer.valueOf(results.getString("posts.postId"));
+            String name = results.getString("users.user_name");
+            String text = results.getString("posts.text");
+            Posts message = new Posts(id, name, text);
+            messages.add(message);
+        }
+        return messages;
+    }
+
+
+    public static void main(String[] args) throws SQLException {
+
+        Server.createWebServer().start();
```

Comment:
This is good so far Vegas.  You still need to add an update method though.  If you add an updateMessage route, updateMessage method that uses SQL queries  and write tests.  You could submit this assignment for Sessions, Sql basics and Sql joins.  But as it stands now you need an update method for it to meet the requirements of Sessions.